neasqc_wp61/models/quantum/embedder.py

```diff
@@ -0,0 +1,316 @@
+
+        Parameters
+        ----------
+        dataset : pd.DataFrame
+            Pandas dataset. Each row of the dataset correspods to a
+            sentence.
+        sentence_embedding : bool
+            States whether we want to produce a sentence embedding or word
+            embedding vector for each sentence in the dataset. True is for
+            sentence embedding, False is for word embedding.
+        dim : int
+            The output dimension of FastText's word/sentence vectors.
+            Default value is 300, but can be set to any value below that.
+        cased: bool
+            Controls whether we casefold our input sentences before
+            vectorising them using FastText. FastText's pretrained model is
+            case sensitive so casefolding will produce different embeddings.
+        """
+        super().__init__(dataset, sentence_embedding)
+        self.dim = dim
+        self.cased = cased
+
+    def compute_embeddings(self) -> DataFrame:
+        """
+        Creates FastText embeddings for sentences in a dataset
+
+        Returns
+        -------
+        embeddings_df : pd.DataFrame
+            A pandas dataframe with the same structure as the original
+            dataset, but with an additional column containing the
+            FastText embeddings corresponding to the sentence contained
+            in each row.
+        """
+        ftu.download_model("en", if_exists="ignore")
+        model = ft.load_model("cc.en.300.bin")
+        if self.dim < 300:
+            ftu.reduce_model(model, self.dim)
+        elif self.dim > 300:
+            print("Error: dimension must be <= 300")
+            return
+
+        embeddings_df = self.dataset.copy()
+        embeddings_df.columns = ["class", "sentence", "sentence_structure"]
+
+        if self.sentence_embedding:
+            print("Generating sentence embeddings. Please wait...")
+            vectorised_sentence_list = []
+            for sentence in embeddings_df.sentence.values:
+                if not self.cased:
+                    sentence = sentence.casefold()
+                sentence_embedding = model.get_sentence_vector(sentence).tolist()
+                vectorised_sentence_list.append(sentence_embedding)
+
+            embeddings_df["sentence_vectorised"] = vectorised_sentence_list
+            print("Done!")
+
+        else:
+            print("Generating word embeddings. Please wait...")
+            vectorised_sentence_list = []
+            for sentence in embeddings_df.sentence.values:
+                word_embeddings_list = []
+                for word in sentence.split():
+                    word_embedding = model.get_word_vector(word).tolist()
+                    word_embeddings_list.append(word_embedding)
+                vectorised_sentence_list.append(word_embeddings_list)
+
+            embeddings_df["sentence_vectorised"] = vectorised_sentence_list
+            print("Done!")
```

Comment:
Remove. This is good when you're devving, not in final code.

Reply:
addressed